src/components/pages/__tests__/vendor.test.tsx

```diff
@@ -16,9 +16,14 @@ describe('<VendorPage/>', () => {
     expect(screen.getByText('Lorem Ipsum')).toBeInTheDocument()
   })
 
-  it('should render the vendors table ', () => {
-    render(withWrappers(<VendorPage />))
-    expect(screen.getByRole('table')).toBeInTheDocument()
+  it('should render the vendors table content properly', async () => {
+    const { container } = render(
+      withWrappers(<VendorPage />, { withRoot: true }),
+    )
+    await waitFor(async () => {
+      expect(screen.queryByTestId('loading-overlay')).not.toBeInTheDocument()
+    })
+    expect(container.innerText).toMatchSnapshot()
```

Comment:
`toMatchInlineSnapshot` will store the snapshot data inside of the function itself

For example, if the snapshot evaluates to `"Hi there, how are you doing?"`, then it'd be something like
```
expect(...).toMatchInlineSnapshot(`"Hi there, how are you doing?"`)
```
This behavior might not be ideal when the snapshot might be large, multiline text, which it might be in this case, so I decided to use `.toMatchSnapshot`, which stores the snapshot in a `__snapshots__` folder instead of storing it inline